CMakeLists.txt

```diff
@@ -203,8 +203,22 @@ if(PCAPPP_USE_DPDK)
 endif()
 
 # Git Commit and Branch
-add_compile_definitions(GIT_COMMIT="${GIT_HASH}")
-add_compile_definitions(GIT_BRANCH="${GIT_BRANCH}")
+find_package(Git)
+if(Git_FOUND)
+  execute_process(
```

Comment:
wondering if the result was your expectation in CI?
```log
-- Found Git: /usr/bin/git (found version "2.39.2") 
-- Building from commit:a67f7c3 on branch:HEAD
```

btw, just curious why you want to add this?

Reply:
It really helps with debugging. For example, since you don't have to use the packaged version of the library and when something is wrong checking the version (for this case Git commit) is really important because the problem might be already fixed with a later version, or you can just check your local version is really up to date.

The other applications/projects can just check and print the version using https://github.com/seladb/PcapPlusPlus/blob/a48ff9219db24e81ac76cf2a11bab796da48652e/Common%2B%2B/src/PcapPlusPlusVersion.cpp#L6-L12 so you can be sure about the version of PcapPlusPlus during runtime.

Reply:
ops, I didn't know we had this file, then it makes sense.

I though we just print the hash during cmake config.

Reply:
Yes except the branch Head but this is due to how checkout is configured in CI.
I think it doesn't matter and I don't want to increase the checkout time as branch information is not really a lot of value